underscore.js

```diff
@@ -1328,7 +1333,10 @@
   // Shortcut function for checking if an object has a given property directly
   // on itself (in other words, not on a prototype).
   _.has = function(obj, key) {
-    return obj != null && hasOwnProperty.call(obj, key);
+    if (!_.isObject(obj)) return false;
+    if (hasOwnProperty.call(obj, key)) return true;
+    // Fix an IE prototype-less object bug. #2308
+    return key in obj && baseGetPrototype(obj) === null;
```

Comment:
I'm willing to believe that this fixes the issue for old IE, but it seems to me that it might also cause false positives in any engine that doesn't have `Object.getPrototypeOf` (including but not limited to old IE).